lib/phoenix/view.ex

```diff
@@ -132,6 +132,7 @@ defmodule Phoenix.View do
 
       @doc "The resource name, as an atom, for this view"
       def __resource__, do: @view_resource
+      def __otp_app__, do: unquote(Mix.Phoenix.otp_app())
```

Comment:
Even though this is compile time, if you want to depend on the otp app, you need to pass it explicitly.